src/components/action-popover/action-popover-menu/action-popover-menu.component.tsx

```diff
@@ -162,6 +168,10 @@ const ActionPopoverMenu = React.forwardRef<
       [focusButton, setOpen, focusIndex, items, setFocusIndex]
     );
 
+    const [caretState, setCaretState] = useState(undefined);
```

Comment:
suggestion: it might be a little more intuitive if we named these a little differently but it's not essential if you're happy with them. I was thinking something like:
```suggestion
    const [childHasSubmenu, setChildHasSubmenu] = useState(false);
     const [childHasIcon, setChildHasIcon] = useState(false);
```